test/integration/scheduler/volume_binding_test.go

```diff
@@ -0,0 +1,405 @@
+			pvs:  []*v1.PersistentVolume{makePV(t, "pv-w-pvc-prebound", classWait, "", "")},
+			pvcs: []*v1.PersistentVolumeClaim{makePVC("pvc-w-prebound", config.ns, &classWait, "pv-w-pvc-prebound")},
+		},
+		"wait pv prebound": {
+			pod:  makePod("pod-w-pv-prebound", config.ns, []string{"pvc-w-pv-prebound"}),
+			pvs:  []*v1.PersistentVolume{makePV(t, "pv-w-prebound", classWait, "pvc-w-pv-prebound", config.ns)},
+			pvcs: []*v1.PersistentVolumeClaim{makePVC("pvc-w-pv-prebound", config.ns, &classWait, "")},
+		},
+		"wait can bind two": {
+			pod: makePod("pod-w-canbind-2", config.ns, []string{"pvc-w-canbind-2", "pvc-w-canbind-3"}),
+			pvs: []*v1.PersistentVolume{
+				makePV(t, "pv-w-canbind-2", classWait, "", ""),
+				makePV(t, "pv-w-canbind-3", classWait, "", ""),
+			},
+			pvcs: []*v1.PersistentVolumeClaim{
+				makePVC("pvc-w-canbind-2", config.ns, &classWait, ""),
+				makePVC("pvc-w-canbind-3", config.ns, &classWait, ""),
+			},
+		},
+		// TODO: this testcase is failing with pod cache modification error
+		// "mix immediate and wait": {
+		//	pod: makePod("pod-mix-bound", config.ns, []string{"pvc-w-canbind-4", "pvc-i-canbind-2"}),
+		//	pvs: []*v1.PersistentVolume{
+		//		makePV(t, "pv-w-canbind-4", classWait, "", ""),
+		//		makePV(t, "pv-i-canbind-2", classImmediate, "", ""),
+		//	},
+		//	pvcs: []*v1.PersistentVolumeClaim{
+		//		makePVC("pvc-w-canbind-4", config.ns, &classWait, ""),
+		//		makePVC("pvc-i-canbind-2", config.ns, &classImmediate, ""),
+		//	},
+		// },
+		// TODO:
+		// immediate mode - PVC cannot bound
+		// wait mode - PVC cannot bind
+		// wait mode - 2 PVCs, 1 cannot bind
+	}
+
+	// create node
+	if _, err := config.client.CoreV1().Nodes().Create(testNode); err != nil {
+		t.Fatalf("Failed to create Node %q: %v", testNode.Name, err)
+	}
+
+	// Create SCs
+	for _, sc := range scs {
+		if _, err := config.client.StorageV1().StorageClasses().Create(sc); err != nil {
+			t.Fatalf("Failed to create StorageClass %q: %v", sc.Name, err)
+		}
+	}
+
+	for name, test := range cases {
+		glog.Infof("Running test %v", name)
+
+		// Create PVs
+		for _, pv := range test.pvs {
+			if _, err := config.client.CoreV1().PersistentVolumes().Create(pv); err != nil {
+				t.Fatalf("Failed to create PersistentVolume %q: %v", pv.Name, err)
+			}
+		}
+
+		// Create PVCs
+		for _, pvc := range test.pvcs {
+			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(pvc); err != nil {
+				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
+			}
+		}
+
+		// Create Pod
+		if _, err := config.client.CoreV1().Pods(config.ns).Create(test.pod); err != nil {
+			t.Fatalf("Failed to create Pod %q: %v", test.pod.Name, err)
+		}
+		if err := waitForPodToSchedule(config.client, test.pod); err != nil {
+			t.Errorf("Failed to schedule Pod %q: %v", test.pod.Name, err)
+		}
+
+		// TODO: Verify PVC/PV binding
+		// TODO: Verify events on Pods and PVCs
```

Comment:
To ensure your test cases are independent and don't conflict, please delete the pod (and probably PVs and PVCs) at the end of each case.

Reply:
done